OstelcoStyles/OstelcoLabel.swift

```diff
@@ -52,13 +52,75 @@ open class OstelcoLabel: UILabel {
 // MARK: - Data labels
 
 public class DataAmountOnHomeLabel: OstelcoLabel {
-    
+
     public override func commonInit() {
         super.commonInit()
         self.appTextColor = .oyaBlue
         self.appFont = OstelcoFont(fontType: .alternateBold,
                                    fontSize: .data)
     }
+    override public var text: String? {
+        didSet {
+            setAttributedText()
+        }
+    }
+    override public var textColor: UIColor? {
+        didSet {
+            setAttributedText()
+        }
+    }
+    override public var font: UIFont? {
+        didSet {
+            setAttributedText()
+        }
+    }
+
+    private func setAttributedText() {
+        attributedText = getStylizeBalanceString(text: self.text ?? "")
+    }
+
+    // Make the string with all the styles required for the balance text
+    // Input text e.g. "54.5 GB"
+    private func getStylizeBalanceString(text: String) -> NSMutableAttributedString {
+        let decimalSeparator: String = Locale.current.decimalSeparator!
+        let bigFont = appFont.toUIFont
+        let smallFont = appFont.toUIFont.withSize(OstelcoFontSize.dataDecimals.toCGFloat)
+        let color = appTextColor.toUIColor
+
+        // Split text to 2 parts, number and units
+        let textArray: [String] = text.components(separatedBy: " ")
+        guard textArray.count >= 2 else {
+            return NSMutableAttributedString(string: text)
+        }
+
+        // Split number string to integer and decimal parts.
+        let numberArray: [String] = textArray[0].components(separatedBy: decimalSeparator)
+        guard numberArray.count >= 1 else {
+            return NSMutableAttributedString(string: text)
+        }
+
+        let integerPart = numberArray[0]
+        // If there is a decimal part.
+        let decimalPart: String? = (numberArray.count >= 2) ? "\(decimalSeparator)\(numberArray[1])": nil
+        let unit = " \(textArray[1])"
+
+        // Add integer part with the big font.
+        let attrString = NSMutableAttributedString(string: integerPart, attributes: [.font: bigFont, .foregroundColor: color])
+        if let decimalPart = decimalPart {
+            // Add decimal part including the decimal character
+            // This portion of text is aligned to top with a smaller font
+            let offset = bigFont.capHeight - smallFont.capHeight
+            let attributes: [NSAttributedString.Key: Any] = [
+                .font: smallFont,
+                .baselineOffset: offset,
+                .foregroundColor: color
+            ]
+            attrString.append(NSMutableAttributedString(string: decimalPart, attributes: attributes))
+        }
+        // Add the modifier part with bigger font.
+        attrString.append(NSMutableAttributedString(string: unit, attributes: [.font: bigFont, .foregroundColor: color]))
+        return attrString
```

Comment:
This is fantastic - you can delete the `DataDecimalsLabel` below since this now covers both the normal amount and the decimals.

Reply:
done